src/Test/TestCombat.java

```diff
@@ -54,6 +58,7 @@ private boolean testUnitDeath()
 
     // Clean up
     testMap.removeUnit(mechA);
+    CombatEngine.modifiers.remove(mod);
```

Comment:
I'm not sure I like needing a separate step to remove modifiers from the engine. That seems like an easy step to overlook, and I'd rather make it more self-contained.